### PR TITLE
1786: set time zone utc

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to ticket

https://leantime.itkdev.dk/dashboard/show?tab=ticketdetails#/tickets/showTicket/1786

#### Description

hmm..

When exporting the file, the date from worked in a mysterious way. If i set timezone UTC, it works as expected. 

#### Screenshot of the result

n/a

### CODE REVIEW

**Comment by sinejespersen** [comment]
@rimi-itk If you find it in your heart to approve this, then just feel free to merge it as well :)

File: Services/AbstractExporter.php [hunk 1]
```diff
@@ -69,18 +69,20 @@ public function export(array $criteria = [], array $options = []): void
...
+     * @return \Carbon\CarbonImmutable|null
      */
-    public function getDateTime(string $value = null, string $default = null): ?Carbon
+    public function getDateTime(string $value = null, string $default = null): ?CarbonImmutable
     {
         if (null !== $value) {
             if ($format = ($_SESSION['usersettings.language.date_format'] ?? null)) {
                 // See https://www.php.net/manual/en/datetimeimmutable.createfromformat.php for details on `!`.
-                return Carbon::createFromFormat('!' . $format, $value) ?: new Carbon($default);
+                $returnDate = CarbonImmutable::createFromFormat('!' . $format, $value) ?: new CarbonImmutable($default);
```
Comment (rimi-itk) [inline Services/AbstractExporter.php]: This line is superfluous and therefore the `if` statement is too.

File: Services/AbstractExporter.php [hunk 2]
```diff
@@ -69,18 +69,20 @@ public function export(array $criteria = [], array $options = []): void
...
+                /* @phpstan-ignore-next-line */
+                return $this->dateTimeHelper->parseUserDateTime($value)->setToDbTimezone();
             }
         }
 
-        return new Carbon($default);
+        return new CarbonImmutable($default);
```
Comment (rimi-itk) [inline Services/AbstractExporter.php]: `$default` should be handled just like `$value`, i.e. passed through the date time helper functions, and then the whole function body can (probably) be rewritten to something like

```php
$value ??= $default;

return null !== $value ? $this->dateTimeHelper->parseUserDateTime($value)->setToDbTimezone() : null;
```

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: Services/AbstractExporter.php (hunk 2 above)
Comment (sinejespersen) [reply Services/AbstractExporter.php]: I've tested it, and it works :)

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**